### ISSUE TITLE
mpirun/horovod not using multiple GPUs

### ISSUE THREAD

**Issue by Losbal** [ludwig-ai/ludwig#1249]
I am trying to run an experiment via the CLI on a single node of a remote HPC cluster with 4 Tesla K80 GPUs, but running nvidia-smi shows that only of the GPUs is actually being utilized, even though I do see 4 PIDs listed for each of the GPUs in the list of processes. I am aware of previous issues with Horovod -- namely [https://github.com/ludwig-ai/ludwig/issues/143](https://github.com/ludwig-ai/ludwig/issues/143) -- but I really thought the command below would check all the boxes, including the instructions to also run mpirun:

mpirun -np 4 --oversubscribe -bind-to none -map-by slot -x LD_LIBRARY_PATH -x PATH ludwig experiment --dataset /home/lfp/Master_Data.csv --data_format csv --experiment_name XP1 --eval_split test --model_name LTT_6 --config_file /home/lfp/LTT_6_config.yaml --output_directory /home/lfp/Out3/ -uh -g 0 1 2 3

I have also included backend: type: horovod in my config file. 

In short, the experiment will run, albeit on only one GPU even though Tensorflow does recognize all 4 CUDA-enabled devices. Any help would be extremely appreciated and many thanks in advance!!

**Comment by tgaddair** [comment]
Hey @Losbal, thanks for reporting this. Let me see if I can reproduce the error. Can you tell me what version of TensorFlow, Horovod, MPI, and Ludwig you're using.

**Comment by tgaddair** [comment]
One thing that stands out is the use of `-g 0 1 2 3`. Can you try removing this and let me know what happens?

**Comment by Losbal** [comment]
Hey @tgaddair - thanks for replying so quickly! Removing `-g 0 1 2 3` appears to have solved it... I can see through nvidia-smi that all 4 GPUs are now being utilized between 88% and 95%. I was going nuts over here and you have no idea how much I appreciate your help - thanks a lot!

**Comment by tgaddair** [comment]
Awesome! Feel free to reach out if anything else comes up.

**Comment by w4nderlust** [comment]
The logic is that the gpus parameter is a parameter assigned for each worker, so if you run without horovod it would assign 4 gpus to one regular worker (not very useful). If you run with horovod, each of the n workers will have 4 gpus assigned. If you don't specify gpus, horovod+Ludwig will figure out the assignment (in this case 1 gpu per worker).
Hopefully this makes sense, and we should probably make ti clearer in the docs.
